### PR TITLE
Anagram

### PR DESCRIPTION
**Pull request by rybmks**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -1,3 +1,4 @@
-pub fn pr(input: &str) {
-    println!("{input}!");
-}
+mod anagram;
+mod clock;
+mod gigasecond;
+mod reverse_string;
```
Comment (YBoy-git) [inline src/lib.rs:4]: You are not exposing the internal functions to the lib user. You must specify `pub mod` to allow users to refer to the module and its child elements.

And then remove `allow(dead_code)` (it shouldn't print any warnings as the functions are going to be counted as public)

File: src/anagram.rs [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+}
+
+fn count_letters(word: &str) -> HashMap<char, usize> {
+    let mut map = HashMap::new();
+    for c in word.chars() {
+        let count = map.entry(c).or_insert(0);
+        *count += 1;
+    }
+    map
+}
```
Comment (YBoy-git) [inline src/anagram.rs:30]: Nice solution.
For me, the most obvious one here was to sort the strings and then compare them, but the solution has a bad time complexity.
As an exercise, write a quick time and space complexity analysis for your solution (as a response to this comment).

**Review by YBoy-git — CHANGES_REQUESTED** [review]

File: src/anagram.rs (hunk 2 above)
Comment (rybmks) [reply src/anagram.rs:30]: O(n * k), where n - amount of strings, k - chars amount.

Comment (YBoy-git) [reply src/anagram.rs:30]: Which complexity is this one? Time or space complexity? Are they the same?

Comment (rybmks) [reply src/anagram.rs:29]: O(n*k) is time complexity. Space complexity is O((m + k )* n), k is amount of unique chars in word, m is word length (when we are creating a lowercase string), n is amount of strings

**Review by YBoy-git — APPROVED** [review]
Nice work on space complexity

**Pull request merged**